tests/unit/services/test_get_profile.py

```
@@ -0,0 +1,105 @@
+        with pytest.raises(
+            ValueError, match="you must specify either auth0_id or user_id"
+        ):
+            service.get_profile()
+
+    def test_get_profile_nonexistent_record(self, test_db):
+        # GIVEN nonexistent record
+        test_db = MagicMock()
+        test_db.query.return_value.fetchone.return_value = None
+
+        # WHEN we call get_profile with nonexistent user
+        result = service.get_profile(auth0_id="invalid")
+
+        # THEN result is None
+        assert result is None
+
+    def test_get_profile_auth0_id(self):
+        # GIVEN an auth0_id
+        auth0_id = "01"
+
+        # WHEN we call get_profile with auth0_id
+        with patch(
+            "policyengine_api.services.user_service.database.query"
+        ) as mock_query:
+            mock_query.return_value.fetchone.return_value = {
+                "auth0_id": auth0_id,
+                "name": "abc",
+            }
+            result = service.get_profile(auth0_id=auth0_id)
+
+        # THEN returns record
+        assert result == {"auth0_id": auth0_id, "name": "abc"}
+        mock_query.assert_called_once_with(
+            "SELECT * FROM user_profiles WHERE auth0_id = ?", (auth0_id,)
+        )
+
+    def test_get_profile_user_id(self):
+        # GIVEN a user_id
+        user_id = "1"
+
+        # WHEN we call get_profile with user_id
+        with patch(
+            "policyengine_api.services.user_service.database.query"
+        ) as mock_query:
+            mock_query.return_value.fetchone.return_value = {
+                "user_id": user_id,
+                "name": "abc",
+            }
+            result = service.get_profile(user_id=user_id)
+
+        # THEN returns record
+        assert result == {"user_id": user_id, "name": "abc"}
+        mock_query.assert_called_once_with(
+            "SELECT * FROM user_profiles WHERE user_id = ?", (user_id,)
+        )
+
+    def test_get_profile_id_datatype(self):
```

__suggestion:__ Remove this test

I believe the `auth0_id` is always a string. If anything, we probably should raise an error here, but I'm not sure if the code currently does that.